src/Identity/IdentityServer/CustomTokenRequestValidator.cs

```diff
@@ -101,6 +108,20 @@ public async Task ValidateAsync(CustomTokenRequestValidationContext context)
             }
         }
 
+        // Attempts to find ssoConfigData for a given validate request subject
+        // this is actually guarenteed to pretty often be null, because more than just sso login requests will come
+        // through here
+        var ssoConfigData = await GetSsoConfigurationDataAsync(context.Result.ValidatedRequest.Subject);
+
+        // You can't put this below the user.MasterPassword != null check because TDE users can still have a MasterPassword
+        // It's worth noting that CurrentContext here will build a user in LaunchDarkly that is anonymous but DOES belong
+        // to an organization. So we will not be able to turn this feature on for only a single user, only for an entire 
+        // organization at a time.
+        if (ssoConfigData != null && _featureService.IsEnabled(FeatureFlagKeys.TrustedDeviceEncryption, CurrentContext))
+        {
+            context.Result.CustomResponse["MemberDecryptionType"] = ssoConfigData.MemberDecryptionType;
```

Comment:
If they do have a Master Password, don't we need to return that as a decryption option here?

Reply:
@jlf0dev Do you mean if they have a master password we force the decryption type to `MasterPassword` or we need to ALSO return whether they have a password here?